cache/cache.go

```diff
@@ -46,6 +46,13 @@ type Cache interface {
 	Name() string
 }
 
+// DeletableCache is cache that can delete its key.
+type DeletableCache interface {
```

Comment:
Do we need this interface? Can't we just add `Delete()` to the `Cache` interface?

Reply:
While I agree, I'd point out that non-compatible API changes are quite painful for updates.

Reply:
Assuming this interface is used with dskit cache client implementations, is it really an issue?

Reply:
Yes, changing `Cache` interface will require a cascading update of multiple dependencies in order to update our internal enterprise product. It is doable (we've done it many times, last time earlier this week), just pointing out the pain.

Reply:
I've seen you introduced `CreateClientWithDelete()` in the last commit. Is that necessary?

Reply:
I agree that we should just add `Delete` to the `Cache` interface. It will require cascading changes but I think they'll be straightforward to fix: the dependent projects will either compile or not, no subtle errors to debug.

Reply:
> I've seen you introduced CreateClientWithDelete() in the last commit. Is that necessary?
It was intended to create DeletableCache without affecting existing Cache interface useage.

Nevertheless I will just update the implementation by adding Delete method to the cache interface.